rubin_scheduler/utils/consdb.py

```diff
@@ -956,6 +989,34 @@ def instrumental_zeropoint_for_band(self, band: str) -> float:
     def site(self):
         return Site("LSST")
 
+    @cached_property
+    def observation_id(self) -> pd.Series:
+        """The unique identifier from the consdb database.
+
+        Returns
+        -------
+        visit_id : `pd.Series`
+            The unique identifier from the consdb database.
+        """
+        if self._have_numeric_values("visit_id"):
+            observation_id = self.consdb_visits["visit_id"]
+        elif self._have_numeric_values("exposure_id"):
+            observation_id = self.consdb_visits["exposure_id"]
+            warn("Cannot use visit_id as observation_id, using exposure_id instead!")
+        elif self._have_numeric_values("day_obs") and self._have_numeric_values("seq_num"):
+            observation_id = self.consdb_visits["day_obs"] * 10000 + self.consdb_visits["seq_num"]
+            warn("Cannot use either visit_id or exposure_id as observation_id, guessing instead!")
+        else:
+            warn(
+                "Cannot use either visit_id, exposure_id, or guess observation_id, just making something up!"
+            )
+            observation_id = pd.Series(
+                Time(self.consdb_visits.obs_start).strftime("9%Y%m%d%H%M%S").astype(int),
+                index=self.consdb_visits.index,
+            )
+
+        return observation_id
```

Comment:
I'm curious - why is this needed? Have some queries from the consDB returned information with non-unique identifiers? 
Although, querying from the exposure table would probably not give unique visit_id values, or at least, it wouldn't be guaranteed to be unique (as there could be two exposures for a given visit).

Reply:
Yes, I had an issue with missing ids that this addressed. 
Because of the way I did the join in the query, there could never be more that one exposure_id matched to a visit, because the join was on exposure_id = visit_id.
With the change I just made, though, exposure_id shouldn't be there any more, so I've removed that branch of the if sequence.